Description: Add debug information to client-server communition
 Add debug call to analyse the lines sent
 from the clien to the server.
 . 
 Useful when debugging communication problems.
Author: Javier Fernández-Sanguino Peña <jfs@debian.org>
Bug-Debian: https://bugs.debian.org/790894
Forwarded: no
Last-Update: <2015-07-04>


--- a/src/firewall.py
+++ b/src/firewall.py
@@ -9,7 +9,7 @@
 import sys
 import os
 import re
-from helpers import log, debug1, debug3, islocal, Fatal, family_to_string, \
+from helpers import log, debug1, debug2, debug3, islocal, Fatal, family_to_string, \
     resolvconf_nameservers
 from fcntl import ioctl
 from ctypes import c_char, c_uint8, c_uint16, c_uint32, Union, Structure, \
@@ -815,6 +815,7 @@
     # sshuttle can launch us as early as possible (and get sudo password
     # authentication as early in the startup process as possible).
     line = sys.stdin.readline(128)
+    debug2('Read line "%s"' % line)
     if not line:
         return  # parent died; nothing to do
 
@@ -823,6 +824,7 @@
         raise Fatal('firewall: expected ROUTES but got %r' % line)
     while 1:
         line = sys.stdin.readline(128)
+        debug2('Read line "%s"' % line)
         if not line:
             raise Fatal('firewall: expected route but got %r' % line)
         elif line == 'GO\n':
@@ -867,6 +869,7 @@
             if do_wait:
                 do_wait()
             line = sys.stdin.readline(128)
+            debug2('Read line "%s"' % line)
             if line.startswith('HOST '):
                 (name, ip) = line[5:].strip().split(',', 1)
                 hostmap[name] = ip
